docs/reference/settings.rst

```diff
@@ -6,6 +6,38 @@ Settings
 
 :ref:`Settings <topics-settings>` for scrapy-zyte-api.
 
+.. setting:: ZYTE_API_AUTO_FIELD_STATS
+
+ZYTE_API_AUTO_FIELD_STATS
+=========================
+
+Default: ``False``
+
+Enables stats that indicate which requested fields come directly from
+:ref:`zyte-api-extract`.
+
+If for any request no page object class is used to override
+:ref:`zyte-api-extract` fields for a given item type, the following stat is
+set:
+
+.. code-block:: python
+
+    "scrapy-zyte-api/auto_fields/<item class import path>": "<space-separated field list>"
+
+If for any request a custom page object class is used to override some
+:ref:`zyte-api-extract` fields, the following stat is set:
+
+.. code-block:: python
+
+    "scrapy-zyte-api/auto_fields/<override class import path>": (
+        "<space-separated list of fields not overridden>"
+    )
+
+.. note:: If that page object class is not a subclass of an ``Auto``-prefixed
+    class from :doc:`zyte-common-items <zyte-common-items:index>`, all fields
+    are assumed to have been overridden, i.e. the stat value is always an empty
+    string.
```

Comment:
Overall, ❤️ this feature! Nice idea @VMRuiz.

I'd like to discuss an alternative implementation, as usual :) What do you think about modifying zyte-common-items auto page classes to set custom field meta - e.g.:

```py
@attrs.define
class AutoArticlePage(BaseArticlePage):
    article: Article

    @field(meta={"auto_field": True})  # TODO: make @auto_field shortcut
    def headline(self) -> Optional[str]:
        return self.article.headline

    @field(meta={"auto_field": True})
    def datePublished(self) -> Optional[str]:
        return self.article.datePublished
```

Then, instead of using mro, etc., just access the field meta. I assume that if you override the field, you use just `@field`, so there is no meta, and we know the field is overritten. 

Also, if you want to override the field, but still indicate that it still doesn't change the data, you can add this meta to a field.

A disadvantage is that it'd require a recent zyte-common-items version to work, and that it's a bit more extensive change.